### ISSUE TITLE
suggestion: mabey a derive-type macro where you can point it to a C jni header and it expands to a function perhaps?

### ISSUE THREAD

**Issue by walksanatora** [mullvad/jnix#52]
this may be a bit complex, but is just a idea i would have liked to see in atleast 1 jni rust crate

**Comment by faern** [comment]
Might be a cool idea. But on our end this crate is mostly in low maintenance mode. It works for our use case and we are currently not planning on spending significant resources on improving it further. I'd love to say "PRs welcome". But I suspect that if the PR is very complex it might take some time for us to get the time to properly review it also.

**Comment by walksanatora** [comment]
yeah this would be a *very* complex PR (Idk how i would even do it, can you define heads of functions in a macro, how do you even read and parse a C header to get the macro function to generate)
